compiler/one-cmds/onelib/utils.py

```diff
@@ -104,44 +106,36 @@ def parse_cfg_and_overwrite(config_path, section, args):
     # TODO support accumulated arguments
 
 
-def parse_cfg(args, driver_name):
-    """parse configuration file. If the option is directly given to the command line,
-       the option is processed prior to the configuration file.
-       That is, if the values parsed from the configuration file already exist in args,
-       the values are ignored."""
-    if is_valid_attr(args, 'config'):
-        config = configparser.ConfigParser()
-        config.optionxform = str
-        config.read(args.config)
-        section_to_run = driver_name
-        # if section is given, use the given section
-        if is_valid_attr(args, 'section'):
-            section_to_run = args.section
-
-        if not config.has_section(section_to_run):
-            raise AssertionError('configuration file must have \'' + section_to_run +
-                                 '\' section')
-
-        if section_to_run == 'one-optimize':
-            # Q. Why we handle default options first?
-            # A. We allow users to overwrite individual options. For example,
-            #    if O1=True and FoldDequantize=False, FoldDequantize is set to False.
-            if 'O1' in config[section_to_run]:
-                for opt in _constant.CONSTANT.O1:
-                    if opt in config[section_to_run]:
-                        continue
-                    setattr(args, opt, True)
-                del config[section_to_run]['O1']
-
-        for key in config[section_to_run]:
-            if is_accumulated_arg(key, driver_name):
-                if not is_valid_attr(args, key):
-                    setattr(args, key, [config[section_to_run][key]])
-                else:
-                    getattr(args, key).append(config[section_to_run][key])
-                continue
+def parse_cfg(config_path: Union[str, None], section_to_parse: str, args, overwrite=True):
+    """
+    parse configuration file and save the information to args
+    
+    :param config_path: path to configuration file
+    :param section_to_parse: section name to parse
+    :param args: object to store the parsed information
+    :param overwrite: flag indicating whether to overwrite the args
+    """
+    if config_path is None:
+        return
+
+    parser = configparser.ConfigParser()
+    parser.optionxform = str
+    parser.read(config_path)
+
+    if not parser.has_section(section_to_parse):
+        raise AssertionError('configuration file must have \'' + section_to_parse +
+                             '\' section')
+
+    for key in parser[section_to_parse]:
+        if is_accumulated_arg(key, section_to_parse):
             if not is_valid_attr(args, key):
-                setattr(args, key, config[section_to_run][key])
+                setattr(args, key, [parser[section_to_parse][key]])
+            else:
+                getattr(args, key).append(parser[section_to_parse][key])
+            continue
+        if not overwrite and hasattr(args, key):
```

Comment:
At this point, `hasattr(args, key)` always returns true (even though it is not set by user). That's because when Python argparser parses arguments with `action='store_true'`, they save attributes name and default value = False.

So, this makes all optimization options in O1.cfg be not applied.

In the test `onecc_044`, `test_onnx_model.onecc_044.opt.cirlce.log` is printed as below.

```
~/ONE/build/release.install/bin/circle2circle test_onnx_model.circle test_onnx_model.onecc_044.opt.circle --convert_nchw_to_nhwc
```

O1 options are not applied. The test is passed because it does not check individual options, but the actual behavior is wrong.

Reply:
A simple workaround would be to use `hasattr(args, key) and getattr(args,key)`, but it adds an assumption that arguments must be defined with `action=store_true. Arguments with `action=store_false` will not work..